dev/com.ibm.ws.microprofile.faulttolerance.2.0/test/src/com/ibm/ws/microprofile/faulttolerance20/impl/MethodResultTest.java

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.faulttolerance20.impl;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+public class MethodResultTest {
+
+    @Test
+    public void testSuccessfulMethodResult() {
+        MethodResult<String> result = MethodResult.success("test");
+        assertFalse("Sucessful result returned true on isFailure", result.isFailure());
+        assertEquals("Sucessful result contained a failure", result.getFailure(), null);
+        assertThat("Sucessful result had the wrong toString", result.toString(), containsString("Method returned value"));
+        assertEquals("Sucessful result had the wrong result", result.getResult(), ("test"));
+    }
+
+    @Test
+    public void testUnsuccessfulMethodResult() {
+        Throwable testThrowable = new NullPointerException("test");
+        MethodResult<String> result = MethodResult.failure(testThrowable);
+        assertTrue("Unsucessful result returned false on isFailure", result.isFailure());
+        assertEquals("Unsucessful result contained the wrong failure", result.getFailure(), testThrowable);
```

Comment:
JUnit's `assertEquals` wants the expected value first and the actual value second. If they're the wrong way round, then the failure message is confusing.

Annoyingly, this is the opposite way round from `assertThat`.

If you prefer, you can change them all to `assertThat(foo, equalTo(bar))`